lib/src/main/scala/spinal/lib/Stream.scala

```diff
@@ -362,7 +362,10 @@ class Stream[T <: Data](val payloadType :  HardType[T]) extends Bundle with IMas
     val rValid = RegNextWhen(self.valid, self.ready) init(False)
     val rData = RegNextWhen(self.payload, if(holdPayload) self.fire else self.ready)
 
-    if (crossClockData) rData.addTag(crossClockDomain)
+    if (crossClockData) {
+      rData.addTag(crossClockDomain)
+      rData.addTag(new crossClockMaxDelay(2, true))
```

Comment:
So m2sPipe crossClockData=true is used in the context where you already have some "arbitration" to tell you when some data cross clock data are stable, and you want to sample them at a safe moment. (small area cross clock)

I would set it to 1 to be safe, as the design around it may assume so. as it cost "nothing"

Reply:
fixed